lib/string.js

```diff
@@ -292,18 +292,76 @@ internals.String = class extends Any {
         });
     }
 
-    guid() {
+    guid(guidOptions) {
 
-        const regex = /^[A-F0-9]{8}(?:-?[A-F0-9]{4}){3}-?[A-F0-9]{12}$/i;
-        const regex2 = /^\{[A-F0-9]{8}(?:-?[A-F0-9]{4}){3}-?[A-F0-9]{12}\}$/i;
+        const brackets = {
+            '{': '}', '[': ']', '(': ')', '': ''
+        };
+
+        const uuids = {
+            'uuidv1': '1',
+            'uuidv2': '2',
+            'uuidv3': '3',
+            'uuidv4': '4',
+            'uuidv5': '5'
+        };
+
+        let versions;
```

Comment:
ok, so are you thinking to move the checks to inside of the test callback? I am sorry for being confused on your thoughts.